docs/standard/design-guidelines/choosing-between-anonymous-and-tuple.md

```diff
@@ -0,0 +1,114 @@
+
+```csharp-interactive
+var dates = new[]
+{
+    DateTime.UtcNow.AddHours(-1),
+    DateTime.UtcNow,
+    DateTime.UtcNow.AddHours(1),
+};
+
+foreach (var anonymous in
+             dates.Select(
+                 date => new { Formatted = $"{date:MMM dd, yyyy hh:mm zzz}", date.Ticks }))
+{
+    Console.WriteLine($"Ticks: {anonymous.Ticks}, formatted: {anonymous.Formatted}");
+}
+```
+
+Anonymous types are instantiated by using the [`new`](../../csharp/language-reference/operators/new-operator.md) operator, and the property names and types are inferred from the declaration. If two or more anonymous object initializers in an assembly specify a sequence of properties that are in the same order and that have the same names and types, the compiler treats the objects as instances of the same type. They share the same compiler-generated type information.
+
+The previous C# snippet projects an anonymous type that would be defined with two properties, much like the following compiler-generated C# class:
+
+```csharp
+internal sealed class f__AnonymousType0
+{
+    public string Formatted { get; }
+    public long Ticks { get; }
+
+    public f__AnonymousType0(string formatted, long ticks)
+    {
+        Formatted = formatted;
+        Ticks = ticks;
+    }
+}
+```
+
+For more information, see [anonymous types](../../csharp/programming-guide/classes-and-structs/anonymous-types.md). The same functionality exists with tuples when projecting into LINQ queries, you can select properties and put them into tuples. These tuples flow through the query, just as anonymous types would. Now consider the following example using the `System.Tuple<string, long>`.
+
+```csharp-interactive
+var dates = new[]
+{
+    DateTime.UtcNow.AddHours(-1),
+    DateTime.UtcNow,
+    DateTime.UtcNow.AddHours(1),
+};
+
+foreach (var tuple in
+            dates.Select(
+                date => new Tuple<string, long>($"{date:MMM dd, yyyy hh:mm zzz}", date.Ticks)))
+{
+    Console.WriteLine($"Ticks: {tuple.Item2}, formatted: {tuple.Item1}");
+}
+```
+
+With the <xref:System.Tuple%602?displayProperty=nameWithType>, the instance exposes numbered item properties, such as `Item1`, and `Item2`. These property names can make it more difficult to understand the intent of the property values, as the property name only provides the ordinal. Furthermore, the `System.Tuple` types are reference `class` types. The <xref:System.ValueTuple%602?displayProperty=nameWithType> however, is a value `struct` type.
+
+```csharp-interactive
+var dates = new[]
+{
+    DateTime.UtcNow.AddHours(-1),
+    DateTime.UtcNow,
+    DateTime.UtcNow.AddHours(1),
+};
+
+foreach (var (formatted, ticks) in
+            dates.Select(
+                date => (Formatted: $"{date:MMM dd, yyyy at hh:mm zzz}", date.Ticks)))
```

Comment:
This would be a really good place to point out the language support for tuples use `System.ValueTuple`. That's more than deconstruct, which you mention below. In addition, tuples support:

- tuple assignment
- tuple equality check.